src/toil/jobStores/azureJobStore.py

```diff
@@ -183,13 +339,26 @@ def delete(self, jobStoreID):
             jobStoreFileID = fileEntity.RowKey
             self.deleteFile(jobStoreFileID)
 
-    def deleteJobStore(self):
-        self.registryTable.delete_entity(row_key=self.namePrefix)
-        self.jobItems.delete_table()
-        self.jobFileIDs.delete_table()
-        self.files.delete_container()
-        self.statsFiles.delete_container()
-        self.statsFileIDs.delete_table()
+    @classmethod
+    def _deleteJobStore(cls, locator):
+        info = cls._processLocator(locator)
+        cls.updateRegistry(info.namePrefix, info.tableService, False)
```

Comment:
If we kept the registry you'd obviously want to updated it last. Otherwise it would be completely pointless.